rest-server/core/src/main/scala/com/actionml/core/spark/SparkLambdaAlgorithm.scala

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright ActionML, LLC under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * ActionML licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.actionml.core.spark
+
+import cats.data.Validated
+import cats.data.Validated.{Invalid, Valid}
+import com.actionml.core.engine.LambdaAlgorithm
+import com.actionml.core.validate.{JsonParser, ValidRequestExecutionError, ValidateError}
+import org.apache.spark.SparkContext
+import org.bson.Document
+
+class SparkLambdaAlgorithm(sparkContextConfig: String) extends LambdaAlgorithm[String] with SparkContextSupport with JsonParser {
+
+  private lazy val validatedSparkContext: Validated[ValidateError, SparkContext] = createSparkContext(sparkContextConfig)
+
+  override def train(): Validated[ValidateError, String] = validatedSparkContext.andThen { sparkContext =>
```

Comment:
I think we talked about the fact that we should not need a special SparkAlgorithm, it should be possible to mix-in a trait to get Spark support, something like:

```scala
class URAlgorithm(...) extends LambdaAlgorith(...) with JsonParser() with SparkSupport(...)
...
```

This way there will be an implicit `sc` for any method that needs it. The params for the context should be created in the factory so they can be passed in to the constructors. 

BTW this is the basis of my bloody merge--doing all the init stuff in the factory.